src/components/GrampsjsFormSelectType.js

```diff
@@ -109,7 +149,8 @@ class GrampsjsFormSelectType extends GrampsjsTranslateMixin(LitElement) {
 
   isValid() {
     const selectType = this.shadowRoot.getElementById('select-type')
-    if (selectType === null) {
+    const customType = this.shadowRoot.getElementById('custom-type') //adding query for custom-type id
+    if (selectType === null && customType === null) { //checking if both types null then return false
```

Comment:
They will only be `null` if the elements don't exist. Is that your intention?

Reply:
My intention was to check if both types were empty:

**if custom type AND select type do not contain input
&nbsp;&nbsp;  return the form is not valid**

Before it seemed that the `isValid()`  function was only checking if select type was empty:

**if select type does not contain input 
&nbsp;&nbsp;  return the form is not valid**

Reply:
Right, but `selectType === null` does not check whether the form contains valid input. The existing code
```python
if (selectType === null) {
      return false
    }
```
is only there because when the element has not fully rendered yet, `#select-type` will not exist in the shadow root yet and checking `selectType.validity.valid` will raise an error.

Reply:
So `isValid()` is for checking if the content has rendered properly. It is called in `checkFormValidity()` from `GrampViewNewEvent.js` in this case. 

There are other elements to check in the form such as date, private, description, but only the `select-type` has a required variable attached to it. Is that the reason we are only checking for the rendering of `select-type`? 

If that is the case then the `custom-type` would also be required if the `select-type` was not being used. Or is the intention here just to check that `select-type` was rendered properly and assume if it was rendered properly the form is valid. Wouldn't we also want to make sure the `custom-type` was rendered properly to consider it a valid form?

To conclude if the intention is just to check if `select-type` was rendered then I would agree that the code to check `custom-type` validity should be removed.